common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -107,7 +115,81 @@ static String getOsPrincipalClassName() {
     }
   }
 
-  // TODO: TACHYON-602 - add TACHYON_LOGIN_MODULE to convert an OS user to Tachyon user.
+  /**
+   * A login module that search the Kerberos or OS user, and then convert to a Tachyon user.
+   */
+  public static class TachyonLoginModule implements LoginModule {
```

Comment:
I am curious why this class `TachyonLoginModule` is a nested class? Here is a quote from "Effective Java" about whether to use nested class:
> A nested class should exist only to serve its enclosing class. If a nested class would be useful in some other context, then it should be a top-level class. 

It seems this class uses `OS_PRINCIPAL_CLASS_NAME` which is private in the enclosing class. But if necessary, we can have a static public method to get this value for the enclosing class.